service/src/main/java/tds/assessment/repositories/impl/AssessmentMapper.java

```diff
@@ -18,82 +18,128 @@
         }
 
-        List<Segment> assessmentSegments = new ArrayList<>();
-        if (segments.isEmpty()) {
-            Segment segment = new Segment(assessmentRow.getString("assessmentSegmentKey"));
-            segment.setAssessmentKey(assessmentRow.getString("assessmentSegmentKey"));
-            updateSegmentWithRowData(segment, assessmentRow, true);
-            assessmentSegments.add(segment);
-        } else {
-            assessmentSegments.addAll(segments.values());
+        return maybeAssessment;
+    }
+
+    /**
+     * Create an {@link tds.assessment.Assessment} from data returned by the query.
+     * <p>
+     *     An assessment must always have one segment.  Since each record returned from the {@code findAssessmentByKey}
+     *     method has enough data to build an assessment and/or a segment (because assessments and segments are stored
+     *     in the same table), this method builds a "default" segment from the same record that is used to build the
+     *     assessment.
+     * </p>
+     *
+     * @param row The {@link tds.assessment.repositories.impl.AssessmentMapper.ResultRow} representing the assessment
+     *            record returned from the query.
+     * @return An {@link tds.assessment.Assessment} with a default {@link tds.assessment.Segment}
+     */
+    private Optional<Assessment> getAssessmentWithDefaultSegment(ResultRow row) {
+        if (row == null) {
+            return Optional.empty();
         }
 
+        // When building the Assessment instance, the "assessmentSegmentKey" field represents the key (unique
+        // identifier) of the Assessment
         Assessment assessment = new Assessment();
-        assessment.setKey(assessmentRow.getString("assessmentSegmentKey"));
-        assessment.setAssessmentId(assessmentRow.getString("assessmentSegmentId"));
-        assessment.setSelectionAlgorithm(assessmentRow.getAlgorithm("selectionalgorithm"));
-        assessment.setStartAbility(assessmentRow.getFloat("startAbility"));
-        assessment.setSubject(assessmentRow.getString("subject"));
-        assessment.setAbilityIntercept(assessmentRow.getFloat("abilityintercept"));
-        assessment.setAbilitySlope(assessmentRow.getFloat("abilityslope"));
-        assessment.setFieldTestStartDate(assessmentRow.getJodaInstantFromTimestamp("ftstartdate"));
-        assessment.setFieldTestEndDate(assessmentRow.getJodaInstantFromTimestamp("ftenddate"));
-        assessment.setAbilitySlope(assessmentRow.getFloat("abilityslope"));
-        assessment.setAbilityIntercept(assessmentRow.getFloat("abilityintercept"));
-        assessment.setAccommodationFamily(assessmentRow.getString("accommodationfamily"));
-        assessment.setMaxOpportunities(assessmentRow.getInt("maxopportunities"));
-        assessment.setInitialAbilityBySubject(assessmentRow.getBoolean("initialabilitybysubject"));
-        assessment.setSegments(assessmentSegments);
+        assessment.setKey(row.getString("assessmentSegmentKey"));
```

Comment:
@millerts, looks like `assessmentRow` was changed to `row` and then the `segments` are created below using the `getSegment` method.